flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/writer/RecordWriterTest.java

```diff
@@ -309,8 +309,6 @@ public void testBroadcastEventBufferReferenceCounting() throws Exception {
 		for (int i = 0; i < queues.length; i++) {
 			assertTrue(parseBuffer(queues[i].remove(), i).isEvent());
 		}
-
-		assertTrue(bufferConsumer.isRecycled());
 	}
```

Comment:
How can you verify your ```bufferConsumer.isRecycled``` logic here if you drop this ?

Reply:
yes, this should not be dropped since this line is the actual test - how about this instead:
```
		// Verify added to all queues
		assertEquals(1, queues[0].size());
		assertEquals(1, queues[1].size());
		
		// get references to buffer consumers (copies from the original event buffer consumer)
		BufferConsumer bufferConsumer1 = queues[0].getFirst();
		BufferConsumer bufferConsumer2 = queues[1].getFirst();

		// process all collected events (recycles the buffer)
		assertTrue(parseBuffer(queues[0].remove(), 0).isEvent());
		assertTrue(parseBuffer(queues[1].remove(), 1).isEvent());

		assertTrue(bufferConsumer1.isRecycled());
		assertTrue(bufferConsumer2.isRecycled());
```